File: python/text/chatgpt.py
from openai import OpenAI

client = OpenAI()

# modelの選択
# gpt-3.5-turbo, gpt-4, gpt-4oから選択
model_name = "gpt-4o"
prompt="こんにちは"

completion = client.chat.completions.create(
  model=model_name,
  messages=[
    {"role": "system", "content": "あなたはミッキーマウスです。必ず日本語で回答してください。"},
    {"role": "user", "content": prompt}
  ]
)

response = completion.choices[0].message.content
print(response)

File: practice/chatgpt.js
import OpenAI from "openai";

const fetchGPT = async (prompt) => {
  const client = new OpenAI(process.env.OPENAI_API_KEY);

  const completion = await client.chat.completions.create({
    model: "gpt-3.5-turbo",
    messages: [
      { role: "system", content: "必ず日本語で回答してください。" },
      { role: "user", content: prompt },
    ],
    n: 1, // 出力の数
    max_tokens: 150, // 出力上限
    temperature: 0.5, // 0.0-1.0
  });

  console.log(completion.choices[0].message.content);

  return completion.choices[0].message.content;
};

const main = async () => {
  console.log("Hello, world!");
};

main();
